### PR TITLE
[api] update chain metrics

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
need to call `SetEvmNetworkID()` when converting protobuf to action

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [x] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/grpcserver.go [hunk 1]
```diff
@@ -336,19 +336,21 @@ func (svr *GRPCServer) SendAction(ctx context.Context, in *iotexapi.SendActionRe
...
+		selp, err := (&action.Deserializer{}).SetEvmNetworkID(svr.coreService.EVMNetworkID()).ActionToSealedEnvelope(in.GetAction())
 		if err != nil {
 			return nil, err
 		}
 		if p, ok := peer.FromContext(ctx); ok {
 			clientIP, _, _ = net.SplitHostPort(p.Addr.String())
 		}
-		if senderAddr, err := address.FromBytes(selp.SrcPubkey().Hash()); err == nil {
-			sender = senderAddr.String()
+		if recipient, ok = selp.Destination(); !ok {
```
Comment (dustinxie) [inline api/grpcserver.go:350]: staking action does not implement `Destination()` interface, so will be `""`

File: api/grpcserver.go [hunk 2]
```diff
@@ -336,19 +336,21 @@ func (svr *GRPCServer) SendAction(ctx context.Context, in *iotexapi.SendActionRe
...
+			recipient = "staking-action"
```
Comment (dustinxie) [inline api/grpcserver.go:351]: staking action does not implement `Destination()` interface, so will be `""`

File: api/grpcserver.go [hunk 3]
```diff
@@ -336,19 +336,21 @@ func (svr *GRPCServer) SendAction(ctx context.Context, in *iotexapi.SendActionRe
 	if in.GetAction().GetCore().GetChainID() > 0 {
 		_apiCallSourceWithChainIDMtc.WithLabelValues(chainID).Inc()
 	} else {
-		var clientIP, sender, recipient string
-		selp, err := (&action.Deserializer{}).ActionToSealedEnvelope(in.GetAction())
+		var (
+			clientIP, recipient string
+			ok                  bool
+		)
+		selp, err := (&action.Deserializer{}).SetEvmNetworkID(svr.coreService.EVMNetworkID()).ActionToSealedEnvelope(in.GetAction())
```
Comment (dustinxie) [inline api/grpcserver.go:343]: need to call `SetEvmNetworkID()` when converting protobuf to action

File: api/grpcserver.go (hunk 2 above)
Comment (Liuhaai) [reply api/grpcserver.go:351]: ToEthTx()

**Review by Liuhaai — APPROVED** [review]

**Review by raullenchai — APPROVED** [review]

**Pull request merged**